app/web/src/newhotness/explore_grid/ExploreGridRow.vue

```diff
@@ -0,0 +1,178 @@
+      "
+      @click.stop.right="
+        (e) =>
+          emit('childClicked', e, component.id, computeIdx(row, columnIndex))
+      "
+    />
+    <!--this fills in any extra spots in an unfilled row-->
+    <div
+      v-for="emptySpot in lanesCount - row.components.length"
+      :key="emptySpot"
+      class="flex-1"
+    />
+  </div>
+</template>
+
+<script lang="ts" setup>
+import clsx from "clsx";
+import { computed, ref } from "vue";
+import * as _ from "lodash-es";
+import {
+  themeClasses,
+  Icon,
+  PillCounter,
+  IconNames,
+  Tones,
+} from "@si/vue-lib/design-system";
+import { tw } from "@si/vue-lib";
+import { ComponentInList } from "@/workers/types/entity_kind_types";
+import { ComponentId } from "@/api/sdf/dal/component";
+import ExploreGridTile from "./ExploreGridTile.vue";
+
+const props = defineProps<{
+  row: ExploreGridRowData;
+  lanesCount: number;
+  focusedComponentId?: ComponentId;
+}>();
+
+const emit = defineEmits<{
+  (e: "childHover", componentId: ComponentId): void;
+  (e: "childUnhover", componentId: ComponentId): void;
+  (
+    e: "childClicked",
+    event: MouseEvent,
+    componentId: ComponentId,
+    componentIdx: number,
+  ): void;
+}>();
+
+interface TitleIcon {
+  iconName: IconNames;
+  iconTone: Tones;
+}
+
+const titleIcon = computed((): TitleIcon | null => {
+  if (props.row.type !== "header") return null;
+
+  switch (props.row.title) {
+    case "Passed":
+      return {
+        iconName: "check-hex-outline",
+        iconTone: "success",
+      };
+    case "Failed":
+      return {
+        iconName: "x-hex-outline",
+        iconTone: "destructive",
+      };
+    default:
+      return null;
+  }
+});
+
+const exploreGridTileRefs = ref<InstanceType<typeof ExploreGridTile>[]>();
+
+const computeIdx = (row: ExploreGridRowData, idx: number) => {
+  if (row.type !== "contentRow") return -1;
+
+  return row.chunkInitialId + idx;
+};
+
+const hoveredId = ref<ComponentId | undefined>(undefined);
+const hover = (componentId: ComponentId, hovered: boolean) => {
+  if (hovered) {
+    hoveredId.value = componentId;
+    emit("childUnhover", componentId);
+  } else {
+    hoveredId.value = undefined;
+    emit("childHover", componentId);
+  }
+};
+
+const tileClasses = (componentId: string) => {
+  const focused = props.focusedComponentId === componentId;
+  if (focused)
+    return themeClasses(tw`border-action-500`, tw`border-action-300`);
+  else if (hoveredId.value === componentId)
+    return themeClasses(tw`border-black`, tw`border-white`);
+  else return "";
+};
+
+defineExpose({ exploreGridTileRefs });
+</script>
+
+<script lang="ts">
+export type ExploreGridRowData =
+  | {
+      type: "contentRow";
+      components: ComponentInList[];
+      chunkInitialId: number;
+      insideSection: boolean;
+    }
+  | {
+      type: "header";
+      title: string;
+      count: number;
+    }
+  | {
+      type: "footer";
+    };
```

Comment:
This is the core of how the PR works: there are three row kinds now when there was previously one.